Extension/src/LanguageServer/settings.ts

```diff
@@ -996,24 +924,46 @@ export class OtherSettings {
         this.resource = resource;
     }
 
-    public get editorTabSize(): number | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<number>("tabSize"); }
-    public get editorInsertSpaces(): boolean | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<boolean>("insertSpaces"); }
-    public get editorAutoClosingBrackets(): string | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<string>("autoClosingBrackets"); }
-    public get filesEncoding(): string | undefined { return vscode.workspace.getConfiguration("files", { uri: this.resource, languageId: "cpp" }).get<string>("encoding"); }
-    public get filesAssociations(): any { return vscode.workspace.getConfiguration("files").get("associations"); }
-    public set filesAssociations(value: any) {
-        void vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Workspace);
+    private getVSCodeSettingAsString(setting: string, configuration: string, resource: any, defaultString?: string): string | undefined {
+        const value = vscode.workspace.getConfiguration(configuration, resource).get<string>(setting);
+        if (isString(value)) {
+            return value;
+        }
+        return defaultString;
+    }
+
+    private getVSCodeSettingAsBoolean(setting: string, configuration: string, resource: any, defaultBoolean?: boolean): boolean | undefined {
+        const value = vscode.workspace.getConfiguration(configuration, resource).get<boolean>(setting);
+        if (isBoolean(value)) {
+            return value;
+        }
+        return defaultBoolean;
     }
+
+    private getVSCodeSettingAsNumber(setting: string, configuration: string, resource: any, defaultNumber?: number): number | undefined {
+        const value = vscode.workspace.getConfiguration(configuration, resource).get<number>(setting);
+        if (isNumber(value)) {
+            return value;
+        }
+        return defaultNumber;
+    }
+
+    // All default values are obtained from the VS Code settings UI. Please update the default values as needed.
+    public get editorTabSize(): number | undefined { return this.getVSCodeSettingAsNumber("editor", "tabSize", this.resource, 4); }
```

Comment:
Programmatically obtain default vs code setting for each setting by calling get on the value obtained from workspace.getConfiguration(), if that is not a valid value, call inspect on the object.